Makefile

```diff
@@ -143,10 +148,82 @@ generate: $(TOOLS_DIR)/stringer
 
 .PHONY: license-check
 license-check:
-	@licRes=$$(for f in $$(find . -type f \( -iname '*.go' -o -iname '*.sh' \) ! -path './vendor/*') ; do \
+	@licRes=$$(for f in $$(find . -type f \( -iname '*.go' -o -iname '*.sh' \) ! -path './vendor/*' ! -path './internal/opentelemetry-proto/*') ; do \
 	           awk '/Copyright The OpenTelemetry Authors|generated|GENERATED/ && NR<=3 { found=1; next } END { if (!found) print FILENAME }' $$f; \
 	   done); \
 	   if [ -n "$${licRes}" ]; then \
 	           echo "license header checking failed:"; echo "$${licRes}"; \
 	           exit 1; \
 	   fi
+
+# Find all .proto files.
+OTEL_PROTO_SUBMODULE := internal/opentelemetry-proto
+PROTO_GEN_DIR        := internal/opentelemetry-proto-gen
+PROTOBUF_TEMP_DIR    := gen/pb-go
+PROTOBUF_SOURCE_DIR  := gen/proto
+ORIG_PROTO_FILES     := $(wildcard $(OTEL_PROTO_SUBMODULE)/opentelemetry/proto/*/v1/*.proto \
+                           $(OTEL_PROTO_SUBMODULE)/opentelemetry/proto/collector/*/v1/*.proto)
+SOURCE_PROTO_FILES   := $(subst $(OTEL_PROTO_SUBMODULE),$(PROTOBUF_SOURCE_DIR),$(ORIG_PROTO_FILES))
+
+ifeq ($(CIRCLECI),true)
+copy-protos-to:
+	# create a dummy container to hold the protobufs
+	docker create -v /defs --name proto-orig alpine:3.4 /bin/true
+	# copy from here into the proto-src directory
+	docker cp ./gen proto-orig:/defs
+
+copy-protobuf-from:
+	rm -fr ./$(PROTO_GEN_DIR)
+	# copy from dummy volume back to our source directory
+	docker cp proto-orig:/defs/$(PROTOBUF_TEMP_DIR)/go.opentelemetry.io/otel/$(PROTO_GEN_DIR) ./$(PROTO_GEN_DIR)
+
+else
+copy-protos-to:
+	@:    # nop
+
+copy-protobuf-from:
+	rm -fr ./$(PROTO_GEN_DIR)
+	mv ./$(PROTOBUF_TEMP_DIR)/go.opentelemetry.io/otel/$(PROTO_GEN_DIR) ./$(PROTO_GEN_DIR)
+endif
+
+
+# This step can be omitted assuming go_package changes are made in opentelemetry-proto repo
+define exec-replace-pkgname
+sed  's+go_package = "github.com/open-telemetry/opentelemetry-proto/gen/go+go_package = "go.opentelemetry.io/otel/internal/opentelemetry-proto-gen+' < $(1) > $(2)
+
+endef
+
+.PHONY: protobuf make-protobufs
+protobuf: protobuf-source copy-protos-to make-protobufs copy-protobuf-from
+
+.PHONY: protobuf-source
+protobuf-source: $(SOURCE_PROTO_FILES) | $(PROTOBUF_SOURCE_DIR)/
+
+# replace opentelemetry-proto v0.4.0 package name by repo-local version
```

Comment:
How does the upgrade path look? Does it include bumping the submodule and going through all the go.mod files?

Also, might not want to include the version in the comment.

```suggestion
# replace opentelemetry-proto package name by repo-local version
```

Reply:
With the new PR #938, it will look as follows:

* Bump git submodule ref at `internal/opentelemetry-proto` to a newer version
* Check in just that change and open the PR.
* The github action from #938 will then run, creating updated `*.pb.go` files (if any) in `internal/opentelemetry-proto-gen` and checking those changes back into the same PR.
* The normal CircleCI build process will run.
* If any changes are required because of the .proto differences, these will need to be resolved by the opener of the PR and will typically be confined to `exporters/otlp/`.